clcache.py

```diff
@@ -747,14 +747,32 @@ def _preprocessToStdout(options):
         return ('E' in options or 'EP' in options) and 'P' not in options
 
     @staticmethod
+    def _parsePathArgument(options, argumentName, sourceFile, extension):
```

Comment:
I think this method name is unfortunate since
* `parsePathArgument` is quite generic (I'd try to add the word `Output` somewhere) and
* the method doesn't actually only parse a path argument, it also does something (inferring an output file name from the given source file name) if the argument is not specified at all.

How about renaming this to `_determineOutputFile`?